### DISCUSSION TITLE
Keyboard navigation/focus blocked when form in modal dialog

### DISCUSSION THREAD

**Discussion by AlexKMarshall** [edmundhung/conform#295] [Q&A]
I'm running conform, zod and Remix with ShadCN/Radix

When I have a form (set to validate onBlur) inside a modal dialog, tabbing through the form is getting stuck. Each time a field loses focus (either through tabbing or just clicking on the next field) focus moves to the dialog itself, rather than to the next field.

I'm guessing this is something to do with the focus trapping implemented in Radix, but I haven't done much digging yet, just wondering if anyone else has encountered and solved this?

Small reproduction in this codesandbox https://codesandbox.io/p/sandbox/magical-spence-vy9wmp
There's 2 identical form pages, one regular, one modal, that show the differences in behavior

**Comment by edmundhung** [comment]
Thanks for the sandbox! I don't have a clue what's the problem yet. I will give it a deeper look once I come back from vacation.

**Reply by dqn** [reply]
@edmundhung 
Any progress on this issue? I would like to know at least a workaround.

**Reply by theskillwithin** [reply]
@edmundhung experiencing this with Radix dialog as  well

**Reply by daniel-a-marshall** [reply]
@edmundhung I'm experiencing this in chrome and it could be because chrome is firing the blur and focusout event when an item is removed from  the dom (https://issues.chromium.org/issues/41484175). I'm not sure it this is relevant or not but might help. Would there be a way to assign the submitter to a hidden element that we mount with the form so it does not have to be added and removed?

**Reply by daniel-a-marshall** [reply]
I added a hidden button 
``` jsx
<button type="submit" className="hidden" data-submitter="true" />
```
and updated the dispatch function on the form context to 
```js
function dispatch(intent) {
    var form = getFormElement();
    var existingSubmitter = form.querySelector('button[data-submitter="true"]');
    var submitter = existingSubmitter || document.createElement('button');
    var buttonProps = getControlButtonProps(intent);
    submitter.name = buttonProps.name;
    submitter.value = buttonProps.value;
    submitter.hidden = true;
    submitter.formNoValidate = true;

    !!existingSubmitter || form === null || form === void 0 || form.appendChild(submitter);
    requestSubmit(form, submitter);
    !!existingSubmitter || form === null || form === void 0 || form.removeChild(submitter);

  } 
```

This works. Not really sure it's the right way to solve the problem. Fairly easy to make a component for that and just drop it in my forms that are inside of radix dialogs
